docs/subnets/commit-reveal.md

```diff
@@ -1,25 +1,57 @@
+
+The weight matrix is public information, and must be, so that the distribution of rewards in the Bittensor platform can be transparently fair. However, this transparency makes it possible for subnet validators to free-ride on the work of other validators by copying the latest consensus rather than independently evaluating subnet miners. This is unfair and potentially degrades the quality of validation work, undermining Bittensor's ability to reward the best miners and produce the best artificial intelligence as a whole.
+
+The commit reveal feature is designed to solve the weight copying problem by giving would-be weight copiers access only to stale weights. Copying stale weights should result in validators departing from consensus. However, it is critical to note that this only works if the consensus weight matrix is constantly changing on the time scale of the commit reveal interval. If the demands on miners are too static, and miner performance is very stable, weight copying will still be successful.
+
+The only solution for this is to continuously change demands on miners, requiring them to continuously evolve their behavior to maintain their scoring. Combined with a properly tuned Commit Reveal interval, this will keep validators honest, as well as producing the best models.
+
+## Commit Reveal and Immunity Period
+
+The [Immunity Period](../glossary.md#immunity-period) is the interval (measured in blocks) during which a miner or validator newly registered on a subnet is 'immune' from deregistration due to performance, because their performance is not yet scored. The duration of this period value should always be larger than the Commit Reveal interval, otherwise the immunity period will expire before a given miner's scores are available, and they may be deregistered without having their work counted.
+
+To ensure that the immunity period is always greater than the Commit Reveal interval, we recommend using the following logic when setting the Immunity Period or Commit Reveal interval hyperparameters for a subnet:
+
+```
+new_immunity_period = (new_commit_reveal_interval - old_commit_reveal_interval) + old_immunity_period
+```
+
+See [Subnet Hyperparameters](./subnet-hyperparameters.md).
+
+
+## Commit reveal in detail
+
+When commit reveal is enabled, it works as follows:  
+
+1. A subnet validator sets the weights normally by using [`set_weights`](pathname:///python-api/html/autoapi/bittensor/core/extrinsics/set_weights/index.html). 
+
+2. Instead of publishing weights openly, an encrypted hash of these consensus weights is committed to the blockchain, using an internal method called [`commit_weights`](pathname:///python-api/html/autoapi/bittensor/core/extrinsics/commit_weights/index.html).
```

Comment:
```suggestion
2. Instead of publishing weights openly, an encrypted copy of these weights is committed to the blockchain, using an internal method called [`commit_weights`](pathname:///python-api/html/autoapi/bittensor/core/extrinsics/commit_weights/index.html).
```